src/html.js

```diff
@@ -80,14 +80,35 @@ export function HTMLString(str) {
 	this.value = str;
 }
 
-// adapted from TiddlyWiki <http://tiddlywiki.com> and Python 3's `html` module
+const HTML_ENCODE_REPLACE = {
+	"&": "&amp;",
+	"<": "&lt;",
+	">": "&gt;",
+	"\"": "&quot;",
+	"'": "&#x27;"
+};
+
+// adapted from html-entities <https://github.com/mdevils/html-entities> (MIT LICENSE)
 export function htmlEncode(str, attribute) {
-	let res = str.replace(/&/g, "&amp;").
-		replace(/</g, "&lt;").
-		replace(/>/g, "&gt;");
-	if(attribute) {
-		res = res.replace(/"/g, "&quot;").
-			replace(/'/g, "&#x27;");
+	const rx = attribute ? /[&<>'"]/g : /[&<>]/g;
+	let res;
+	let match = rx.exec(str);
+	if(match) {
+		res = "";
+		let lastIndex = 0;
+		do {
+			if(lastIndex !== match.index) {
+				res += str.substring(lastIndex, match.index);
+			}
+			res += HTML_ENCODE_REPLACE[match[0]];
+			lastIndex = rx.lastIndex;
+		} while((match = rx.exec(str)));
+
+		if(lastIndex !== str.length) {
+			res += str.substring(lastIndex);
+		}
+	} else {
+		res = str;
```

Comment:
Personally, I prefer the [bouncer pattern](http://rikschennink.nl/thoughts/the-bouncer-pattern/) to avoid excessive nesting, reduce action at a distance etc.

```javascript
if(!match) {
    return str;
}

let rx = attribute ? /[&<>'"]/g : /[&<>]/g;
…
return res;
```

Reply:
Hmm.. yeah it would better match the nature of the function. If it would be able to speak it would say:
> I have seen so many strings in my life, i expect there are no evill in it and then it's no more todo. Just take it.
> But when i found something, then i have to do a bit more to protect you.
😄

Reply:
> This looks pretty solid!

Btw. I have to thank. You have created an amazing project (i really love JSX) and spend your time to let me part of it ❤️

Reply:
I'm glad you find it useful and are willing to contribute.

I should mention that I'm not currently using complate myself and thus might not always be able to provide a timely response. So just to be safe, you might wanna send a quick note (e.g. by creating an issue) before starting any larger efforts. Having said that, there are other people around here who are perfectly capable of providing feedback.